library/Icingadb/Redis/VolatileStateResults.php

```diff
@@ -36,11 +36,17 @@ class VolatileStateResults extends ResultSet
     /** @var string Object type service */
     protected const TYPE_SERVICE = 'service';
 
+    /** @var array|null Columns to be selected if they were explicitly set, if null all columns are selected */
+    protected ?array $columns = null;
+
     public static function fromQuery(Query $query)
     {
         $self = parent::fromQuery($query);
         $self->resolver = $query->getResolver();
         $self->redisUnavailable = Backend::getRedis()->isUnavailable();
+        if (! empty($query->getColumns())) {
+            $self->columns = $query->getColumns();
+        }
```

Comment:
I suppose it's fine to just assign the property here. Later on you can still differentiate, as whether null or an empty array does not make a difference, does it?